File: models/pantry_item.py
from . import db
from datetime import datetime

class PantryItem(db.Model):
    __tablename__ = 'pantry_items'
    
    item_id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.user_id'), nullable=False)
    category_id = db.Column(db.Integer, db.ForeignKey('categories.category_id'))
    name_vi = db.Column(db.String(200), nullable=False)
    name_en = db.Column(db.String(200))
    quantity = db.Column(db.Numeric(10, 2), nullable=False)
    unit = db.Column(db.String(20), nullable=False)
    purchase_date = db.Column(db.Date)
    expiry_date = db.Column(db.Date)
    location = db.Column(db.Enum('fridge', 'freezer', 'pantry'), default='fridge')
    add_method = db.Column(db.Enum('manual', 'barcode', 'ocr'), default='manual')
    barcode = db.Column(db.String(50))
    image_url = db.Column(db.String(500))
    notes = db.Column(db.Text)
    status = db.Column(db.Enum('active', 'used', 'expired', 'deleted'), default='active')
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    def __repr__(self):
        return f'<PantryItem {self.name_vi}>'

File: dotnet_backend/BepTroLy.API/Models/PantryItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BepTroLy.API.Models;

[Table("pantry_items")]
public class PantryItem
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("item_id")]
    public int ItemId { get; set; }

    [Column("user_id")]
    public int UserId { get; set; }

    [Column("category_id")]
    public int? CategoryId { get; set; }

    [Required]
    [MaxLength(200)]
    [Column("name_vi")]
    public string NameVi { get; set; } = string.Empty;

    [MaxLength(200)]
    [Column("name_en")]
    public string? NameEn { get; set; }

    [Column("quantity", TypeName = "decimal(10, 2)")]
    public decimal Quantity { get; set; }

    [Required]
    [MaxLength(20)]
    [Column("unit")]
    public string Unit { get; set; } = string.Empty;

    [Column("purchase_date")]
    public DateOnly? PurchaseDate { get; set; }

    [Column("expiry_date")]
    public DateOnly? ExpiryDate { get; set; }

    // Enum: 'fridge', 'freezer', 'pantry'
    [Column("location")]
    public string Location { get; set; } = "fridge";

    // Enum: 'manual', 'barcode', 'ocr'
    [Column("add_method")]
    public string AddMethod { get; set; } = "manual";

    [MaxLength(50)]
    [Column("barcode")]
    public string? Barcode { get; set; }

    [MaxLength(500)]
    [Column("image_url")]
    public string? ImageUrl { get; set; }

    [Column("notes", TypeName = "text")]
    public string? Notes { get; set; }

    // Enum: 'active', 'used', 'expired', 'deleted'
    [Column("status")]
    public string Status { get; set; } = "active";

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    [ForeignKey("UserId")]
    public User? User { get; set; }

    [ForeignKey("CategoryId")]
    public Category? Category { get; set; }
}
